src/lib/unlang/xlat_eval.c

```diff
@@ -222,6 +222,121 @@ static inline void xlat_debug_log_result(request_t *request, fr_value_box_t cons
 	RDEBUG2("  --> %pV", result);
 }
 
+/** Process list of boxed values provided as input to an xlat
+ *
+ * Ensures that the value boxes passed to an xlat function match the
+ * requirements listed in its "args", and escapes any tainted boxes
+ * using the specified escaping routine.
+ *
+ * @param[in] ctx		in which to allocate any buffers.
+ * @param[in,out] in		value boxes provided as input.
+ * @param[in] input_type	required by xlat.
+ * @param[in] args		definition of arguments required by xlat.
+ */
+static xlat_action_t xlat_process_args(TALLOC_CTX *ctx, fr_value_box_list_t *in, request_t *request,
+					xlat_input_type_t input_type, xlat_arg_parser_t args[])
+{
+	xlat_arg_parser_t	*arg;
+	fr_value_box_t		*vb, *next;
+
+	/*
+	 *	No args registered for this xlat
+	 */
+	if (!args) return XLAT_ACTION_DONE;
+
+	switch (input_type) {
+	case XLAT_INPUT_UNPROCESSED:
+		/*
+		 *	xlat needs no input processing just return.
+		 */
+		return XLAT_ACTION_DONE;
+
+	case XLAT_INPUT_MONO:
+		/*
+		 *	xlat takes all input as a single vb.
+		 */
+		vb = fr_dlist_head(in);
+		if (vb) {
```

Comment:
A lot of nesting here.  Maybe

```c
if (!vb && args->required) {
missing:
    RERROR("Missing required argument");
    return XLAT_ACTION_FAIL;
}
```

makes it a bit easier to follow the code too.

Reply:
Isn't the the same level of nesting?

Reply:
It lets you unindent the code between lines 260-277

Reply:
https://wiki.freeradius.org/contributing/coding-standards#code-style_return-early :)